src/Microsoft.Sbom.Common/Attributes/DirectoryAttribute.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Sbom.Common.Config.Attributes;
+
+/// <summary>
+/// This attribute is used to mark a property as a directory.
+/// </summary>
+[AttributeUsage(AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+public sealed class DirectoryAttribute : Attribute
```

Comment:
Technically this is any path. I would rename this to FilePath or just Path attribute (you probably will have to check with name collision)